TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+    }
+    
+    [task setValuesForKeysWithDictionary:[NSDictionary dictionaryWithDictionary:dictionaryTask]];
+    
+    NSError *savingError;
+    
+    @try {
+        
+        if ([self.managedObjectContext save:&savingError]){
+            NSLog(@"Tasks saved Ok");
+        }else{
+            NSLog(@"error %@", savingError);
+        }
+        
+    } @catch (NSException *exception) {
+        
+        NSLog(@"exception: %@", [exception debugDescription]);
+        
+    }
+
+}
+
+
+-(NSArray *)getTasks{
+    
+    NSArray *resultArray;
+    NSMutableArray *fetchedTasks = [[NSMutableArray alloc] init];
+    Tasks *task;
+    
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Tasks" inManagedObjectContext:self.managedObjectContext];
+    
+    NSNumber *status = [NSNumber numberWithUnsignedInteger:3];
+
+    NSPredicate *predicateLid = [NSPredicate predicateWithFormat:@"status < %@", status];
+    
+    NSPredicate *compPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:[NSArray arrayWithObjects:predicateLid, nil]];
+    [fetchRequest setPredicate:compPredicate];
+    
+    NSInteger sortOrder = [[NSUserDefaults standardUserDefaults] integerForKey:@"sortOrder"];
+
+    NSSortDescriptor *sortingPredicate;
+
+    
+    switch (sortOrder) {
+        case 0:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"title" ascending:YES];
+            break;
+        }
+        case 1:{
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"title" ascending:NO];
+            break;
+        }
+        case 2:{
+
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"priority" ascending:YES];
+            break;
+        }
+        case 3:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"priority" ascending:NO];
+            break;
+        }
+        case 4:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"dateto" ascending:YES];
+            break;
+        }
+        case 5:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"dateto" ascending:NO];
+            break;
+        }
+
+    }
+
+    NSArray *arrSortOrder = [[NSArray alloc] initWithObjects:sortingPredicate, nil];
+    fetchRequest.sortDescriptors = arrSortOrder;
+
+    
+    NSError *requestError = nil;
+    
+    [fetchRequest setIncludesPendingChanges:YES];
+    [fetchRequest setEntity:entity];
+    
+    NSArray *items = [self.managedObjectContext executeFetchRequest:fetchRequest error:&requestError];
+    
+    NSLog(@"records in Tasks: %ld", (long) [items count]);
+
+    for (task in items) {
+
+        NSArray *keys = [[[task entity] attributesByName] allKeys];
+        NSMutableDictionary *fetchedTask = [NSMutableDictionary dictionaryWithDictionary:[task dictionaryWithValuesForKeys:keys]];
```

Comment:
It's unclear why the fetchedTask object isn't passed around the stack and dicts are used instead. By not doing this you'd save lots of coding.